config/main.py

```diff
@@ -3245,7 +3247,7 @@ def add_snmp_agent_address(ctx, agentip, port, vrf):
             click.echo("ManagementVRF is Enabled. Provide vrf.")
             return False
     found = 0
-    ip = ipaddress.ip_address(agentip)
+    ip = ipaddress.ip_address(agentip.split('%')[0])
```

Comment:
>ip [](http://example.com/codeflow?start=4&length=2)

You split twice. You can move this statement earlier, and check `is_ipaddress(ip)`.

Reply:
updated as suggested